src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.ValidateSignature.cs

```diff
@@ -208,40 +211,36 @@ private static SignatureValidationResult ValidateSignatureWithKey(
             CryptoProviderFactory cryptoProviderFactory = validationParameters.CryptoProviderFactory ?? key.CryptoProviderFactory;
             if (!cryptoProviderFactory.IsSupportedAlgorithm(jsonWebToken.Alg, key))
             {
-                return new SignatureValidationResult(
-                    ValidationFailureType.SignatureValidationFailed,
-                    new ExceptionDetail(
-                        new MessageDetail(
-                            LogMessages.IDX14000,
-                            LogHelper.MarkAsNonPII(jsonWebToken.Alg),
-                            key),
-                        ExceptionDetail.ExceptionType.SecurityTokenInvalidAlgorithm,
-                        new StackFrame()));
+                return new ExceptionDetail(
+                    new MessageDetail(
+                        TokenLogMessages.IDX10400,
+                        LogHelper.MarkAsNonPII(jsonWebToken.Alg),
+                        key),
+                    ExceptionType.SecurityTokenInvalidAlgorithm,
+                    new StackFrame(true));
             }
 
-            AlgorithmValidationResult result = validationParameters.AlgorithmValidator(
+            Result<string, ExceptionDetail> result = validationParameters.AlgorithmValidator(
                 jsonWebToken.Alg,
                 key,
                 jsonWebToken,
                 validationParameters,
                 callContext);
-            if (!result.IsValid)
-                return new SignatureValidationResult(
-                    ValidationFailureType.SignatureValidationFailed,
-                    result.ExceptionDetail);
+
+            if (!result.IsSuccess)
```

Comment:
null check?

Reply:
Value type explained in another comment.